packages/core/content-type-builder/server/src/services/builder.ts

```diff
@@ -1,18 +1,30 @@
 export const getReservedNames = () => {
   return {
-    models: ['boolean', 'date', 'date-time', 'dateTime', 'time', 'upload'],
+    // use kebab case everywhere since singularName and pluralName are validated that way
+    models: [
+      'boolean',
+      'date',
+      'date-time',
+      'time',
+      'upload',
+      'then', // https://github.com/strapi/strapi/issues/15557
+      'rest', // https://github.com/strapi/strapi/issues/13643
```

Comment:
Can you add `subscription` with a reference to this issue: https://github.com/strapi/strapi/issues/18638

Reply:
> Can you add `subscription` with a reference to this issue: #18638

From reading that issue, I can't in v4 because it works for anyone not using graphql. I will add it to the list for v5 (or try to solve it before then)